### ISSUE TITLE
API documentation and usage examples

### ISSUE THREAD

**Issue by curtwagner1984** [Dusk-Labs/nightfall#13]
While this is an internal library, it would be nice to have some sort of API documentation that details what the library can do and what API endpoints should be used to achieve its functionality. 

Something like a short 'Getting Started' guide that gives a basic usage example and basic API documentation that tells arguments and outputs are expected to the core API entry points.

**Comment by Apezdr** [comment]
I agree, would be nice to have documentation to lean on if someone wanted to prototype the transcode engine for another use case.
